ApeWorX/ape-solidity
# fix: issue where sometimes showed unused import remappings in settings

### What I did

Allow getting import remappings to filter by the given source paths.
This allows us to remove unused remappings from settings if possible.

### How I did it

Read the lines of the source files and only use remappings from config if present

### How to verify it

* Have a bunch of remappings in config e.g.

```yaml
solidity:
  import_remapping:
    - "@openzeppelin/contracts=OpenZeppelin/4.6.0"
    - "@openzeppelin=OpenZeppelin/4.6.0"
    - "@solidity-bytes-utils/contracts=solidity-bytes-utils/v0.8.0"
```

* Have a source file only use one of them e.g.:

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.2;
import "@openzeppelin/contracts/<something>";
contract Contract {
    address public owner;
}
```

Check settings and make sure only the 1 shows:

```python
project.extract_manifest().compilers[0].settings["remappings"]
```

### Checklist
